### PR TITLE
Fix : the event must never serialize null or empty fields

### PR DESCRIPTION
**Pull request by rxlabz**
## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description

new checks to be sure to not serialize null or empty fields

## :green_heart: How did you test it?
unit tests

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: dart/lib/src/protocol/message.dart [hunk 1]
```diff
@@ -20,10 +20,20 @@ class Message {
...
-    };
+    final json = <String, dynamic>{};
+
+    if (formatted != null && formatted.isNotEmpty) {
+      json['formatted'] = formatted;
+    }
+
+    if (template != null && template.isNotEmpty) {
+      json['message'] = template;
+    }
```
Comment (marandaneto) [inline dart/lib/src/protocol/message.dart]: we don't need to `isNotEmpty` for `Strings` but only for collections

File: dart/lib/src/protocol/sentry_event.dart [hunk 2]
```diff
@@ -316,18 +320,24 @@ class SentryEvent {
...
-          'version': sdkVersion,
-        };
+    Map<String, dynamic> sdkMap;
+    if (sdk != null && (sdkMap = sdk.toJson()).isNotEmpty) {
+      json['sdk'] = sdkMap;
+    } else {
+      json['sdk'] = <String, String>{
+        'name': sdkName,
+        'version': sdkVersion,
+      };
```
Comment (marandaneto) [inline dart/lib/src/protocol/sentry_event.dart]: I'd remove this just for the sake of DRY, `_prepareEvent` already sets it.

File: dart/lib/src/protocol/message.dart (hunk 1 above)
Comment (rxlabz) [reply dart/lib/src/protocol/message.dart]:
```suggestion
    if (formatted != null) {
      json['formatted'] = formatted;
    }

    if (template != null) {
      json['message'] = template;
    }
```

File: dart/lib/src/protocol/sentry_event.dart (hunk 2 above)
Comment (rxlabz) [reply dart/lib/src/protocol/sentry_event.dart]: ✅

**Pull request merged**